api/controllers/ReferenceController.js

```diff
@@ -195,18 +194,16 @@ module.exports = {
     });
   },
 
-  approve: function (req, res) {
-    Reference.findOne(req.allParams().id, function (err, ref) {
-      if (!ref || err) {
-        return res.notFound(err);
+  approve: async function (req, res) {
+    try {
+      var ref = await Reference.findOne(req.allParams().id);
+      if (!References.isApprovable(ref)) {
+        return res.badRequest();
       }
-      References.approve(ref, req.allParams().approve).then(function (result) {
-        return res.ok(result);
-      }, function (error) {
-        console.log(error);
-        return res.serverError(error);
-      });
-    });
+      return await References.approve(ref, req.allParams().approve);
+    } catch (err) {
+      return res.serverError(err);
+    }
```

Comment:
Refactored this, it's now async. I also added an `isApprovable` function in one of the services. It's mainly used to make the ejs files a bit more compact, but we also use it here to verify that the request to approve something is valid.